# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/tests/gtest/TestMozPromise.cpp
# Commit: 48eaa84c206d
# Full Hash: 48eaa84c206dbdda637c57b55ddbd07e975ba103
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-06-23 16:15:06
# Regressor Bug: 1637500
# File Overlap Count: 2
# Description:
#   Bug 1637500 - P2. Rename methods as they are not always dealing with "threads". r=froydnj
#   
#   Before P1, GetCurrentThreadSerialEventTarget would have always returned the same data as NS_GetCurrentThread, making the comment incorrect Now it will properly return the running TaskQueue if any.
#   
#   This change of name more clearly exposes what they are doing, as we aren't always dealing with threads directly; but a nsISerialEventTarget
# ==============================================================================

diff -r 00687d06a8a6 -r 48eaa84c206d xpcom/tests/gtest/TestMozPromise.cpp
--- a/xpcom/tests/gtest/TestMozPromise.cpp	Tue Jun 23 05:02:39 2020 +0000
+++ b/xpcom/tests/gtest/TestMozPromise.cpp	Tue Jun 23 05:05:36 2020 +0000
@@ -432,7 +432,7 @@
 {
   TestPromise::CreateAndResolve(42, __func__)
       ->Then(
-          GetCurrentThreadSerialEventTarget(), __func__,
+          GetCurrentSerialEventTarget(), __func__,
           [](int aResolveValue) -> void { EXPECT_EQ(aResolveValue, 42); },
           DO_FAIL);
 
@@ -457,7 +457,7 @@
   RefPtr<TestPromise> promise1 = TestPromise::CreateAndResolve(42, __func__);
   RefPtr<TestPromise::Private> promise2 = new TestPromise::Private(__func__);
   promise2->Then(
-      GetCurrentThreadSerialEventTarget(), __func__,
+      GetCurrentSerialEventTarget(), __func__,
       [&](int aResolveValue) -> void { EXPECT_EQ(aResolveValue, 42); },
       DO_FAIL);
 
@@ -476,7 +476,7 @@
   promise->Resolve(42, __func__);
   EXPECT_EQ(value, false);
   promise->Then(
-      GetCurrentThreadSerialEventTarget(), __func__,
+      GetCurrentSerialEventTarget(), __func__,
       [&](int aResolveValue) -> void {
         EXPECT_EQ(aResolveValue, 42);
         value = true;
@@ -492,7 +492,7 @@
       new TestPromiseExcl::Private(__func__);
   promise->UseSynchronousTaskDispatch(__func__);
   promise->Then(
-      GetCurrentThreadSerialEventTarget(), __func__,
+      GetCurrentSerialEventTarget(), __func__,
       [&](int aResolveValue) -> void {
         EXPECT_EQ(aResolveValue, 42);
         value = true;
@@ -511,9 +511,9 @@
   // For direct task dispatch to be working, we must be within a
   // nested event loop. So the test itself must be dispatched within
   // a task.
-  GetCurrentThreadSerialEventTarget()->Dispatch(
+  GetCurrentSerialEventTarget()->Dispatch(
       NS_NewRunnableFunction("test", [&]() {
-        GetCurrentThreadSerialEventTarget()->Dispatch(
+        GetCurrentSerialEventTarget()->Dispatch(
             NS_NewRunnableFunction("test", [&]() {
               EXPECT_EQ(value1, true);
               value2 = true;
@@ -525,7 +525,7 @@
         promise->Resolve(42, __func__);
         EXPECT_EQ(value1, false);
         promise->Then(
-            GetCurrentThreadSerialEventTarget(), __func__,
+            GetCurrentSerialEventTarget(), __func__,
             [&](int aResolveValue) -> void {
               EXPECT_EQ(aResolveValue, 42);
               EXPECT_EQ(value2, false);
@@ -547,9 +547,9 @@
   // For direct task dispatch to be working, we must be within a
   // nested event loop. So the test itself must be dispatched within
   // a task.
-  GetCurrentThreadSerialEventTarget()->Dispatch(
+  GetCurrentSerialEventTarget()->Dispatch(
       NS_NewRunnableFunction("test", [&]() {
-        GetCurrentThreadSerialEventTarget()->Dispatch(
+        GetCurrentSerialEventTarget()->Dispatch(
             NS_NewRunnableFunction("test", [&]() {
               EXPECT_EQ(value1, true);
               value2 = true;
@@ -562,7 +562,7 @@
         EXPECT_EQ(value1, false);
         promise1
             ->Then(
-                GetCurrentThreadSerialEventTarget(), __func__,
+                GetCurrentSerialEventTarget(), __func__,
                 [&](int aResolveValue) -> RefPtr<TestPromise> {
                   EXPECT_EQ(aResolveValue, 42);
                   EXPECT_EQ(value2, false);
@@ -574,7 +574,7 @@
                 },
                 DO_FAIL)
             ->Then(
-                GetCurrentThreadSerialEventTarget(), __func__,
+                GetCurrentSerialEventTarget(), __func__,
                 [&](int aResolveValue) -> void {
                   EXPECT_EQ(aResolveValue, 43);
                   EXPECT_EQ(value2, false);
@@ -596,9 +596,9 @@
   // For direct task dispatch to be working, we must be within a
   // nested event loop. So the test itself must be dispatched within
   // a task.
-  GetCurrentThreadSerialEventTarget()->Dispatch(
+  GetCurrentSerialEventTarget()->Dispatch(
       NS_NewRunnableFunction("test", [&]() {
-        GetCurrentThreadSerialEventTarget()->Dispatch(
+        GetCurrentSerialEventTarget()->Dispatch(
             NS_NewRunnableFunction("test", [&]() {
               EXPECT_EQ(value1, true);
               value2 = true;
@@ -611,7 +611,7 @@
         RefPtr<TestPromise::Private> promise2 =
             new TestPromise::Private(__func__);
         promise2->Then(
-            GetCurrentThreadSerialEventTarget(), __func__,
+            GetCurrentSerialEventTarget(), __func__,
             [&](int aResolveValue) -> void {
               EXPECT_EQ(aResolveValue, 42);
               EXPECT_EQ(value2, false);